modules/ROOT/pages/define-configurations-and-connection-providers.adoc

```diff
@@ -85,6 +85,44 @@ Therefore, the getAccessToken() method *must* be invoked on the state object for
 
 This consideration is especially important when using third-party client libraries that might perform caching.
 
+=== Don't use the access token in the connection creation
+
+The creation of the connection object in the ConnectionProvider#connect method must not depend on the presence of a valid access token. Even if the connection needs to use the token to perform a one-time only request, that should not be done in the creation of the connection. The request must be deferred to when the connection is first used.
+
+To this end, the org.mule.runtime.api.util.LazyValue class can be leveraged. So that when a value is first used, the action of retrieving the value is done, and later uses will get the value that was cached from the first time.
```

Comment:
```suggestion
To adhere to this practice, you can use the `org.mule.runtime.api.util.LazyValue` class. When a value is initially used, the value is retrieved; subsequent uses get the value that was cached initially.
```

Reply:
I do not really understand which "first time" this references, so if we need to modify this further, please let me know.

Reply:
The "first time" references to the moment when the value is first used. You create something that knows how to resolve a value, when you use it for the first time the value is resolved, later usages will return the value that was given before without executing the resolution again.

Reply:
To adhere to this practice, you can use the `org.mule.runtime.api.util.LazyValue` class. After the value is retrieved for the first time, subsequent uses get the same value that was cached in the initial retrieval.